src/averageAges.js

```diff
@@ -37,7 +41,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(person => (
+    withChildren
+      ? people.find(child => child.mother === person.name)
+      : person.sex === 'f'
```

Comment:
Here as well, let's create the variable for `person.sex === 'f'` expression, cause you are using that value twice.

Reply:
didn't get it, maybe I'm some sort of blind, but I think that  person.sex === 'f' expression used only once in the whole task, isn't it?